src/BroadcastMsg.chpl

```diff
@@ -30,15 +31,15 @@ module BroadcastMsg {
     }
     const segs = toSymEntry(gs, int);
     // Check that values exists (can be any dtype)
-    const gv = getGenericTypedArrayEntry(valName, st);
+    const gv = getGenericTypedArrayEntry(msgArgs.getValueOf("valName"), st);
     // Name of result array
     const rname = st.nextName();
     // This operation has two modes: one uses a permutation to reorder the answer,
     // while the other does not
-    const usePerm: bool = usePermStr.toLower() == 'true';
+    const usePerm: bool = msgArgs.getValueOf("permute").toLower() == 'true';
```

Comment:
is it possible to use `getBoolValue` here?

Reply:
@pierce314159 you are right. Initially I had an issue casting to `bool` because I forgot to run `toLower` on the string before the cast. I updated `getBoolValue` to run `toLower` and everything is good to go now using `getBoolValue`.